src/lsdb/catalog/dataset/healpix_dataset.py

```diff
@@ -104,20 +104,55 @@ def _repr_divisions(self):
         divisions = pd.Index(self.get_ordered_healpix_pixels(), name=name)
         return divisions
 
-    def _create_modified_hc_structure(self, **kwargs) -> HCHealpixDataset:
+    def _create_modified_hc_structure(
+        self, hc_structure=None, updated_schema=None, **kwargs
+    ) -> HCHealpixDataset:
         """Copy the catalog structure and override the specified catalog info parameters.
 
         Returns:
             A copy of the catalog's structure with updated info parameters.
         """
-        return self.hc_structure.__class__(
-            catalog_info=self.hc_structure.catalog_info.copy_and_update(**kwargs),
-            pixels=self.hc_structure.pixel_tree,
-            catalog_path=self.hc_structure.catalog_path,
-            schema=self.hc_structure.schema,
-            moc=self.hc_structure.moc,
+        if hc_structure is None:
+            hc_structure = self.hc_structure
+        return hc_structure.__class__(
+            catalog_info=hc_structure.catalog_info.copy_and_update(**kwargs),
+            pixels=hc_structure.pixel_tree,
+            catalog_path=hc_structure.catalog_path,
+            schema=hc_structure.schema if updated_schema is None else updated_schema,
+            moc=hc_structure.moc,
         )
 
+    def _create_updated_dataset(
+        self,
+        ddf: nd.NestedFrame | None = None,
+        ddf_pixel_map: DaskDFPixelMap | None = None,
+        hc_structure: HCHealpixDataset | None = None,
+        updated_catalog_info_params: dict | None = None,
+    ) -> Self:
+        """Creates an"""
+        if ddf is None:
+            ddf = self._ddf
```

Comment:
```suggestion
        ddf = ddf or self._ddf
```

And in a half-dozen places following.

Maybe even update the docstring to say that this method exists to coalesce user-provided arguments with object members.

Reply:
Oh actually it seems like a nonempty ddf raises an error rather than being true. I've replaced this with an inline if.